plans/init.pp

```diff
@@ -127,6 +127,7 @@
   # them to the peadm_nodes group and agent_nodes
   $inventory.each |$k, $v| { $v.each |$target| {
     Target.new($target.merge($target_config)).add_to_group('peadm_nodes')
+    wait_until_available($target['name'], wait_time => 300)
```

Comment:
How about instead of waiting once for each target, we just do a single wait, at the point in the plan where we actually need the targets to be ready? This lets the plan finish the rest of its calculations without blocking, potentially revealing other problems to the use faster if there are any.

I would suggest on line ~200, add it there. We can just wait for the group 'peadm_nodes'. The parameter is a TargetSpec, so group names will work here.

```diff
   unless $stage {
+    wait_until_available('peadm_nodes', wait_time => 300)
+
     # Once all the infrastructure data has been collected, handoff to puppetlabs/peadm
     run_plan('peadm::install', $params + $extra_peadm_params)
```

Reply:
We could also put it above the `unless $stage` conditional. That would let the plan validate that targets were available before returning, even if `$stage == true`